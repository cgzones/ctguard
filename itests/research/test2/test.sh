#!/bin/sh

set -eu

cleanup () {
	rm -f test.output
}

cleanup


OUTPUT=$(../../../ctguard-research --cfg-file test.conf --input -f << EOF
Sep 24 12:10:03 desktopdebian unix_chkpwd[12490]: password check failed for user (christian)
Sep 24 12:10:03 desktopdebian unix_chkpwd[12490]: password check failed for user (root)
Sep 24 12:10:03 desktopdebian unix_chkpwd[12490]: password check failed for user (christian)
quit
EOF
)
echo "$OUTPUT" > test.output

echo "Comapring expected vs actual output:"
diff -u test.output.expected test.output

cleanup

echo "\n\n\nsuccess!!!\n\n"
