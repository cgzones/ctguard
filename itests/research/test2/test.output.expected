ctguard-research

  Enter loginput to see its handling.
   - use 'quit' to exit

>
  Processing 'Sep 24 12:10:03 desktopdebian unix_chkpwd[12490]: password check failed for user (christian)' ...
    Format (#1) ...  syslog matching
      Traits:
        control -> ##false##
        format -> ##syslog##
        hostname -> ####
        source_domain -> ####
        source_program -> ####
        time_scanned -> ##Thu Jan  1 00:00:00 1970 UTC##
        time_send -> ##Thu Jan  1 00:00:00 1970 UTC##
      Extracted fields:
        domain -> ##unix_chkpwd##
        hostname -> ##desktopdebian##
        log -> ##password check failed for user (christian)##
        timestamp -> ##Sep 24 12:10:03##
      Groups: ####
      Priority: 0
    Processing rules (#1) level 1
    Checking level 1 rule 1 ...field 'domain' matching...|trait 'format' matching...|full match: true(true && true)
    Matching level 1 rule: 1
    Level 1 rule fit: 1
    Checking child rules (#1) ...
    Processing rules (#1) level 2
    Checking level 2 rule 2 ...regex match|full match: true(true && true)
    Matching level 2 rule: 1
    Level 2 rule fit: 2
    Checking child rules (#0) ...
    Processing rules (#0) level 3
    Matching level 3 rule: 0
      Traits:
        control -> ##false##
        format -> ##syslog##
        hostname -> ####
        source_domain -> ####
        source_program -> ####
        time_scanned -> ##Thu Jan  1 00:00:00 1970 UTC##
        time_send -> ##Thu Jan  1 00:00:00 1970 UTC##
      Extracted fields:
        domain -> ##unix_chkpwd##
        hostname -> ##desktopdebian##
        log -> ##password check failed for user (christian)##
        timestamp -> ##Sep 24 12:10:03##
        username -> ##christian##
      Groups: ##authentication_failure system##
      Priority: 4
    Checking group rules...
    Processing rules (#1) level 1
    Checking level 1 rule 3 ...init rstate|full match: false(true && false)
    Matching level 1 rule: 0
      Traits:
        control -> ##false##
        format -> ##syslog##
        hostname -> ####
        source_domain -> ####
        source_program -> ####
        time_scanned -> ##Thu Jan  1 00:00:00 1970 UTC##
        time_send -> ##Thu Jan  1 00:00:00 1970 UTC##
      Extracted fields:
        domain -> ##unix_chkpwd##
        hostname -> ##desktopdebian##
        log -> ##password check failed for user (christian)##
        timestamp -> ##Sep 24 12:10:03##
        username -> ##christian##
      Groups: ##authentication_failure system##
      Priority: 4
  End extracting.

Rule matched:
	rule id  : 2
	priority : 4
	message  : sshd: failed user login
Would create alert with priority 4.

>
  Processing 'Sep 24 12:10:03 desktopdebian unix_chkpwd[12490]: password check failed for user (root)' ...
    Format (#1) ...  syslog matching
      Traits:
        control -> ##false##
        format -> ##syslog##
        hostname -> ####
        source_domain -> ####
        source_program -> ####
        time_scanned -> ##Thu Jan  1 00:00:00 1970 UTC##
        time_send -> ##Thu Jan  1 00:00:00 1970 UTC##
      Extracted fields:
        domain -> ##unix_chkpwd##
        hostname -> ##desktopdebian##
        log -> ##password check failed for user (root)##
        timestamp -> ##Sep 24 12:10:03##
      Groups: ####
      Priority: 0
    Processing rules (#1) level 1
    Checking level 1 rule 1 ...field 'domain' matching...|trait 'format' matching...|full match: true(true && true)
    Matching level 1 rule: 1
    Level 1 rule fit: 1
    Checking child rules (#1) ...
    Processing rules (#1) level 2
    Checking level 2 rule 2 ...regex match|full match: true(true && true)
    Matching level 2 rule: 1
    Level 2 rule fit: 2
    Checking child rules (#0) ...
    Processing rules (#0) level 3
    Matching level 3 rule: 0
      Traits:
        control -> ##false##
        format -> ##syslog##
        hostname -> ####
        source_domain -> ####
        source_program -> ####
        time_scanned -> ##Thu Jan  1 00:00:00 1970 UTC##
        time_send -> ##Thu Jan  1 00:00:00 1970 UTC##
      Extracted fields:
        domain -> ##unix_chkpwd##
        hostname -> ##desktopdebian##
        log -> ##password check failed for user (root)##
        timestamp -> ##Sep 24 12:10:03##
        username -> ##root##
      Groups: ##authentication_failure system##
      Priority: 4
    Checking group rules...
    Processing rules (#1) level 1
    Checking level 1 rule 3 ...inactive_s(1/2)|full match: false(true && false)
    Matching level 1 rule: 0
      Traits:
        control -> ##false##
        format -> ##syslog##
        hostname -> ####
        source_domain -> ####
        source_program -> ####
        time_scanned -> ##Thu Jan  1 00:00:00 1970 UTC##
        time_send -> ##Thu Jan  1 00:00:00 1970 UTC##
      Extracted fields:
        domain -> ##unix_chkpwd##
        hostname -> ##desktopdebian##
        log -> ##password check failed for user (root)##
        timestamp -> ##Sep 24 12:10:03##
        username -> ##root##
      Groups: ##authentication_failure system##
      Priority: 4
  End extracting.

Rule matched:
	rule id  : 2
	priority : 4
	message  : sshd: failed user login
Would create alert with priority 4.

>
  Processing 'Sep 24 12:10:03 desktopdebian unix_chkpwd[12490]: password check failed for user (christian)' ...
    Format (#1) ...  syslog matching
      Traits:
        control -> ##false##
        format -> ##syslog##
        hostname -> ####
        source_domain -> ####
        source_program -> ####
        time_scanned -> ##Thu Jan  1 00:00:00 1970 UTC##
        time_send -> ##Thu Jan  1 00:00:00 1970 UTC##
      Extracted fields:
        domain -> ##unix_chkpwd##
        hostname -> ##desktopdebian##
        log -> ##password check failed for user (christian)##
        timestamp -> ##Sep 24 12:10:03##
      Groups: ####
      Priority: 0
    Processing rules (#1) level 1
    Checking level 1 rule 1 ...field 'domain' matching...|trait 'format' matching...|full match: true(true && true)
    Matching level 1 rule: 1
    Level 1 rule fit: 1
    Checking child rules (#1) ...
    Processing rules (#1) level 2
    Checking level 2 rule 2 ...regex match|full match: true(true && true)
    Matching level 2 rule: 1
    Level 2 rule fit: 2
    Checking child rules (#0) ...
    Processing rules (#0) level 3
    Matching level 3 rule: 0
      Traits:
        control -> ##false##
        format -> ##syslog##
        hostname -> ####
        source_domain -> ####
        source_program -> ####
        time_scanned -> ##Thu Jan  1 00:00:00 1970 UTC##
        time_send -> ##Thu Jan  1 00:00:00 1970 UTC##
      Extracted fields:
        domain -> ##unix_chkpwd##
        hostname -> ##desktopdebian##
        log -> ##password check failed for user (christian)##
        timestamp -> ##Sep 24 12:10:03##
        username -> ##christian##
      Groups: ##authentication_failure system##
      Priority: 4
    Checking group rules...
    Processing rules (#1) level 1
    Checking level 1 rule 3 ...active_s(2/2)|full match: true(true && true)
    Matching level 1 rule: 1
    Level 1 rule fit: 3
    Checking child rules (#0) ...
    Processing rules (#0) level 2
    Matching level 2 rule: 0
      Traits:
        control -> ##false##
        format -> ##syslog##
        hostname -> ####
        source_domain -> ####
        source_program -> ####
        time_scanned -> ##Thu Jan  1 00:00:00 1970 UTC##
        time_send -> ##Thu Jan  1 00:00:00 1970 UTC##
        trigger_same_logs -> ##Sep 24 12:10:03 desktopdebian unix_chkpwd[12490]: password check failed for user (christian)
Sep 24 12:10:03 desktopdebian unix_chkpwd[12490]: password check failed for user (christian)
##
      Extracted fields:
        domain -> ##unix_chkpwd##
        hostname -> ##desktopdebian##
        log -> ##password check failed for user (christian)##
        timestamp -> ##Sep 24 12:10:03##
        username -> ##christian##
      Groups: ##authentication_failure system##
      Priority: 8
  End extracting.

Rule matched:
	rule id  : 3
	priority : 8
	message  : Multiple failed authentications for same user.
Would create alert with priority 8.

>
Ending
